lib/Base/Organization/Thread.vala

```diff
@@ -27,11 +27,36 @@ using GLib;
  * "main post". It will display all posts preceding the main post until
  * the top one, as well as all replies to the main post.
  */
-public abstract class Backend.Thread : Backend.Collection {
+public abstract class Backend.Thread : Collection<Object>, PullableCollection {
+
+  /**
+   * The session handling calls for this collection.
+   */
+  public Session session { get; construct; }
 
   /**
    * The post from which the thread is build.
    */
   public Post main_post { get; construct; }
 
+  /**
+   * The id of the newest item in the collection.
+   */
+  protected string? newest_item_id { get; set; default = null; }
+
+  /**
+   * Run at construction of the class.
+   */
+  construct {
+    // FIXME: Use a sorter usable for Thread.
+    sorter = new ReversePostSorter ();
+  }
```

Comment:
Does this need to be a class? Could we define a delegate signature and make them functions?

I'm using a lot of Python day-to-day and I much prefer the approach of making things functions when they don't need the weight of being objects 🙂

Reply:
That ties into the issue with "sorting threads". Currently, the compare method can look up and down when a post is in a thread, so posts can be sorted according to the start post. 

However, sorter is a [`CompareDataFunc`](https://valadoc.org/glib-2.0/GLib.CompareDataFunc.html), which compares two items without their context. Therefore I wanted to use the class to allow for additional context.\
But since then I found [`GLib.Sequence.sort_iter`](https://valadoc.org/glib-2.0/GLib.Sequence.sort_iter.html), which could allow for that context. Or we find a better way to sort with threads in mind.

Reply:
Sounds like it could be computationally expensive, but definitely more flexible. I've not looked at that code, though.

Just passing a `SequenceIterCompareFunc` reference around (without a class) does seem like the better option. Anything using objects because you can hold state on them most likely wouldn't be thread safe.